File: lab_02/lab2.cpp
#include <iostream>
#include <chrono> // модуль для подсчета времени
#include <Windows.h> // Нужно для SetConsoleOutputCP

using namespace std;

double f(double x) {
    return x * x - x * x + x*4 - x*5 + x + x;
}

int time() {

    int n;
    cout << "enter the num of iterations: ";

    if (cin >> n) {
        
        if (n < 0) {
            cout << "ERROR: your num < 0\n";
            return 1;
        }

        double x = 5;

        auto start = chrono::high_resolution_clock::now();

        for (int i = 0; i < n; i++) {
            f(x);
        }

        auto end = chrono::high_resolution_clock::now();

        auto duration_us = chrono::duration_cast<chrono::microseconds>(end - start);

        cout << "time: " << duration_us.count() << " \u00B5s\n";

        return 0;

    }

    else {

        cout << "ERROR: restart and enter uint num!!\n";
        return 1;

    }
}

int main() {

    SetConsoleOutputCP(65001);

    char s;
    int t;

    do {
        t = time();
        if (t == 0) {
            cout << "run again? (y/n): "; 
            cin >> s;
        }
    } while (s == 'y' && t == 0);
    
}

File: lab_02/lab2.py
import time

def f(x):
    return x * x - x * x + x*4 - x*5 + x + x

def time1():

    print(f'enter the num of iterations: ', end='')
    n = input()

    if n.isdigit():

        n = int(n)

        if n >= 0:

            x = 5
            start = time.time()

            for i in range(n):
                m = f(x)

            end = time.time()

            execution_time = end - start
            print(f'time: {execution_time*10**6} us')

        else:
            print('ERROR: your num < 0\n')

    else:
        print('ERROR: restart and enter uint num!!\n')

s = 'y'

while s == 'y':
    time1()
    print('run again? (y/n): ', end='')
    s = input()


